core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -263,21 +261,15 @@ public void handleItemPickup(final PlayerPickupItemEvent event) {
 
   @EventHandler
   public void lockFireTick(final MatchLoadEvent event) {
-    oldFireTick = Boolean.parseBoolean(event.getMatch().getWorld().getGameRuleValue(DO_FIRE_TICK));
     event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, Boolean.toString(false));
   }
 
   // if no doFireTick <gamerule> is found, then use doFireTick setting saved in world
   @EventHandler
   public void unlockFireTick(final MatchStartEvent event) {
-    ImmutableMap<String, String> gameRules =
-        event.getMatch().getModule(GameRulesMatchModule.class).getGameRules();
-    String newFireTick = gameRules.get(DO_FIRE_TICK);
-    if (!gameRules.isEmpty() && newFireTick != null) {
-      event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, newFireTick);
-    } else {
-      event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, Boolean.toString(oldFireTick));
-    }
+    String farts = event.getMatch().getModule(GameRulesMatchModule.class).getMapGameRules().get(DO_FIRE_TICK);
+    String newFireTick = event.getMatch().getModule(GameRulesMatchModule.class).getGameRules().get(DO_FIRE_TICK);
+    event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, newFireTick != null ? newFireTick : farts);
```

Comment:
```suggestion
    String fireTick = event.getMatch().needModule(GameRulesMatchModule.class).getGameRule(DO_FIRE_TICK);
    event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, fireTick);
```
Once you create a single method that does the logic, you'll be able to use this. You must change getModule to needModule, because you will otherwise NPE if for some reason it was null